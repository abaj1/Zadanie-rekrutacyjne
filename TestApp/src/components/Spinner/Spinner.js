import React from 'react';
import {View, StyleSheet} from 'react-native';
import {ActivityIndicator, Colors} from 'react-native-paper';

const Spinner = ({style, isLoading}) => {
  return (
    <View style={{...styles.container, ...style}}>
      <ActivityIndicator animating={isLoading} color={Colors.red800} />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    position: 'absolute',
    width: '100%',
    height: '100%',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
    zIndex: 1000,
    backgroundColor: 'rgba(0,0,0,0.6)',
  },
});

export default Spinner;
