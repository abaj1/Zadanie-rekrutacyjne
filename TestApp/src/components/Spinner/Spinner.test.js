import React from "react";
import Spinner from "../../components/Spinner/Spinner";
import renderer from "react-test-renderer";

describe("Spinner", () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });
  it("Should render Spinner when state is undefined ", () => {
    const spinner = renderer.create(<Spinner></Spinner>).toJSON();

    expect(spinner).toMatchSnapshot();
  });
});
