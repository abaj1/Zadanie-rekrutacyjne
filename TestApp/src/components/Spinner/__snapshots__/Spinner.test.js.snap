// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Should render Spinner when state is undefined  1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(0,0,0,0.6)",
      "bottom": 0,
      "flex": 1,
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": "100%",
      "zIndex": 1000,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": 64,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#0768fd",
          "borderRadius": 2,
          "height": 4,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 4,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#9e54b0",
          "borderRadius": 2,
          "height": 4,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 4,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#df216d",
          "borderRadius": 2,
          "height": 4,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 4,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#ff9300",
          "borderRadius": 2,
          "height": 4,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
          "width": 4,
        }
      }
    />
  </View>
</View>
`;
